### PR TITLE
Fix TS component towers

### LINKED ISSUE #9428: TS component towers are broken
Astor reported 2 issues with the TS component towers (discovered in his own mod):
As it turns out, 3 weeks ago 21186c10b6a37b72402229ca0ac03444eea27dd3 broke the component tower's make animation (possibly for other structures as well).
And 9da56f51e21064088f29e14d90839c27e10f240a broke their firing.

I was unable to find any other broken actors.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/26935950-ts-component-towers-are-broken?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by penev92**
As the commit message suggests, `WithWallSpriteBody` was hijacking `WithSpriteBody`'s `DefaultAnimation` before/during the "make" animation, and the `Building` trait is counting on the make animation to finish and notify it that the building is complete. But that never happens, so the building is "stuck" with `Building.BuildComplete == false`, which then causes all sorts of trouble.

Fixes #9428.

### CODE REVIEW

**Comment by obrakmann** [comment]
Not the first time that the `MakeAnimation`<->`BuildComplete` dependency has caused bugs.

:+1:

File: OpenRA.Mods.Common/Traits/Render/WithWallSpriteBody.cs [hunk 1]
```diff
@@ -131,8 +133,26 @@ static void UpdateNeighbours(Actor self)
 
 		public void AddedToWorld(Actor self)
 		{
-			DefaultAnimation.PlayFetchIndex(NormalizeSequence(self, Info.Sequence), () => adjacent);
-			UpdateNeighbours(self);
+			var building = self.Trait<Building>();
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Render/WithWallSpriteBody.cs]: `Requires<BuildingInfo>` too please.

File: OpenRA.Mods.Common/Traits/Render/WithWallSpriteBody.cs [hunk 2]
```diff
@@ -131,8 +133,26 @@ static void UpdateNeighbours(Actor self)
...
+				DefaultAnimation.PlayFetchIndex(NormalizeSequence(self, Info.Sequence), () => adjacent);
+				UpdateNeighbours(self);
+			}
+			else
+			{
+				// Wall adjacency rendering has lower priority.
+				// Wait for the "make" animation to finish first instead of overriding DefautAnimation.
+				self.QueueActivity(Util.SequenceActivities(
+					new WaitFor(() => building.BuildComplete),
+					new CallFunc(() =>
```
Comment (pchote) [inline OpenRA.Mods.Common/Traits/Render/WithWallSpriteBody.cs]: I believe we have a notification for when the make animation finishes - this should live there instead.

Comment (pchote) [inline OpenRA.Mods.Common/Traits/Render/WithWallSpriteBody.cs]: It's the same one that sets `building.BuildComplete` true.

Comment (penev92) [inline OpenRA.Mods.Common/Traits/Render/WithWallSpriteBody.cs]: Yes, the one one `WithMakeAnimation`. I can't introduce such an ugly dependency. :X

Comment (pchote) [inline OpenRA.Mods.Common/Traits/Render/WithWallSpriteBody.cs]: I'm talking about `INotifyBuildComplete`, which we use in a bunch of the other render traits.

**Comment by penev92** [comment]
Well derp. This sure is simpler.

**Comment by abcdefg30** [comment]
Looks good to me. :+1: / :white_check_mark:

**Pull request merged**